R-Interface/R/symlinkTools.R

```diff
@@ -0,0 +1,302 @@
+
+# Goal of this script:
+# Go through all the module folders
+# Collect all the symlinks
+# Know renv-cache path
+# Subtract renv-cache path from the symlinks to go to relative paths
+# Unix:
+#   Recreate the symlinks directly on unix but relative this time
+# Win:
+#   Save them to a file that can be used to restore them on Windows
+#   Have a function here or in that file that restores them and add it as a custom action to Wix
+
+pastePath <- function(path) { return(paste0(  path, collapse=.Platform$file.sep)     ) }
+splitPath <- function(path) 
+{ 
+  path  <- gsub("\\", "/", path, fixed=TRUE)
+  paths <- strsplit(path, "/")[[1]]; 
+  return(paths[paths != ""]) # Remove "empty" dir between // like: "blablac//weird/path"
+}
+
+#This returns atwo functions that can be used to convert paths in two direcitons
+determineOverlap <- function(targetRoot, sourceRoot)
+{
+  targetSplit <- splitPath(targetRoot)
+  sourceSplit <- splitPath(sourceRoot)
+  len         <- min(length(targetSplit), length(sourceSplit))
+  overlap     <- 0
+
+  for(idx in seq(len))
```

Comment:
if `len` is ever 0 then you get a loop over 1, 0. 0 will cause issues since you use it as an index. With `seq_len` you'd get an empty vector whenever `len` is 0 and the loop would never be entered.
```suggestion
  for(idx in seq_len(len))
```